plutus-benchmark/flat/Report.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Report ( Size(..)
+              , Measures
+              , measureSize
+              , withRatio
+              , showContractMeasures
+              , showCodecSizes
+              ) where
+
+import qualified Data.ByteString        as BS
+import           Data.Function          (on)
+import           Data.List              (sort)
+import           Data.Map               (Map)
+import qualified Data.Map               as Map
+import           Data.Text              (Text, pack, unpack)
+import           GHC.Generics
+import           Text.PrettyPrint.Boxes
+
+import           Prelude                as P
+
+data Size = Size
+  { szContract :: Text
+  , szCodec    :: Text
+  , szBytes    :: Integer
+  } deriving (Eq, Show, Generic)
+
+type Measures = Map Text [Size]
+
+-- | Order by contract name and serialized contract size.
+instance Ord Size where
+  compare = (compare `on` szContract) P.<> (compare `on` szBytes)
+
+measureSize ::
+     Map (Text, Text) BS.ByteString
+  -> Map Text [Size]
+measureSize =
+  Map.map sort .
+  Map.foldlWithKey go Map.empty
+  where
+    go :: Map Text [Size] -> (Text, Text) -> BS.ByteString -> Map Text [Size]
+    go ms (contractName, codecName) contract =
+      let oldMs = Map.findWithDefault [] contractName ms
+          sizeMs = Size { szContract = contractName
+                        , szCodec    = codecName
+                        , szBytes    = fromIntegral $ BS.length contract
+                        }
+          newMs = sizeMs : oldMs
+      in Map.insert contractName newMs ms
```

Comment:
I don't know how you feel about lens, but `ms & at contractName <>~ sizeMs` would probably do what you want ("modify the value at this index by `mappend`ing on an extra bit")

Reply:
I am not very used to `lens` yet, but comments like this are bound to make me a fan.